apps/dh/api-dh/source/DataHub.WebApi/Modules/MarketParticipant/Actor/ActorOperations.cs

```diff
@@ -13,16 +13,105 @@
 // limitations under the License.
 
 using Energinet.DataHub.WebApi.Clients.MarketParticipant.v1;
-using Energinet.DataHub.WebApi.GraphQL.Types;
-using Energinet.DataHub.WebApi.GraphQL.Types.Actor;
-using Energinet.DataHub.WebApi.GraphQL.Types.Process;
+using Energinet.DataHub.WebApi.Extensions;
+using Energinet.DataHub.WebApi.Modules.MarketParticipant.Actor.Models;
 
-namespace Energinet.DataHub.WebApi.GraphQL.Mutation;
+namespace Energinet.DataHub.WebApi.Modules.MarketParticipant.Actor;
 
-public partial class Mutation
+public static partial class ActorOperations
 {
+    [Query]
+    public static Task<ActorDto> GetSelectedActorAsync(
+            [Service] IHttpContextAccessor httpContextAccessor,
+            [Service] IMarketParticipantClient_V1 client)
+    {
+        var user = httpContextAccessor.HttpContext?.User;
+        var associatedActor = user?.GetAssociatedActor()
+            ?? throw new InvalidOperationException("No associated actor found.");
+
+        return client.ActorGetAsync(associatedActor);
+    }
+
+    [Query]
+    public static async Task<ActorDto> GetActorByIdAsync(
+            Guid id,
+            [Service] IMarketParticipantClient_V1 client) =>
+            await client.ActorGetAsync(id);
+
+    [Query]
+    public static async Task<IEnumerable<ActorDto>> GetActorsAsync(
+            [Service] IMarketParticipantClient_V1 client) =>
+            await client.ActorGetAsync();
+
+    [Query]
+    public static async Task<IEnumerable<ActorDto>> GetActorsByOrganizationIdAsync(
+            Guid organizationId,
+            [Service] IMarketParticipantClient_V1 client) =>
+            await client.OrganizationActorAsync(organizationId);
+
+    [Query]
+    public static async Task<IEnumerable<ActorDto>> GetActorsForEicFunctionAsync(
+            EicFunction[]? eicFunctions,
+            [Service] IMarketParticipantClient_V1 client) =>
+            (await client.ActorGetAsync())
+            .Where(x => eicFunctions != null && eicFunctions.Contains(x.MarketRole.EicFunction));
+
+    [Query]
+    public static async Task<AssociatedActors> GetAssociatedActorsAsync(
+        string email,
+        [Service] IMarketParticipantClient_V1 client)
+    {
+        var user = (await client.UserOverviewUsersSearchAsync(
+                1,
+                int.MaxValue,
+                UserOverviewSortProperty.Email,
+                SortDirection.Asc,
+                new UserOverviewFilterDto
+                {
+                    UserStatus = [],
+                    UserRoleIds = [],
+                })).Users
+            .FirstOrDefault(x => string.Equals(email, x.Email, StringComparison.OrdinalIgnoreCase));
```

Comment:
suggestion: not for now, but would it be that hard for backend to add a "GetByEmail" endpoint? 🙂

Reply:
I will ask our backenders